### ISSUE TITLE
Moderation - WSOD when submitting a report

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#13277]
- [ X] I have searched open and closed issues for duplicates

I don't know if this should already work or if it is simply not finished yet. But I think I should leave this here in case this is a bug.

### Bug Description

When submitting a report a white screen of death appears. There are several error messages in the log.

```
2023-07-11T17:56:19Z index [WARNING]: E_WARNING: Undefined array key "category" {"code":2,"message":"Undefined array key \"category\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":186} - {"file":"Create.php","line":186,"function":"pickCategory","request-id":"64ad97c31543f","uid":"c0aa2c","process_id":2150242}
2023-07-11T17:56:19Z index [WARNING]: E_WARNING: Undefined array key "category" {"code":2,"message":"Undefined array key \"category\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":187} - {"file":"Create.php","line":187,"function":"pickCategory","request-id":"64ad97c31543f","uid":"c0aa2c","process_id":2150242}
2023-07-11T17:56:19Z index [WARNING]: E_WARNING: Undefined array key "category" {"code":2,"message":"Undefined array key \"category\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":188} - {"file":"Create.php","line":188,"function":"pickCategory","request-id":"64ad97c31543f","uid":"c0aa2c","process_id":2150242}
2023-07-11T17:56:19Z index [WARNING]: E_WARNING: Undefined array key "category" {"code":2,"message":"Undefined array key \"category\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":189} - {"file":"Create.php","line":189,"function":"pickCategory","request-id":"64ad97c31543f","uid":"c0aa2c","process_id":2150242}
2023-07-11T17:56:19Z index [WARNING]: E_WARNING: Undefined array key "category" {"code":2,"message":"Undefined array key \"category\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":190} - {"file":"Create.php","line":190,"function":"pickCategory","request-id":"64ad97c31543f","uid":"c0aa2c","process_id":2150242}
2023-07-11T17:56:19Z index [WARNING]: E_WARNING: Undefined array key "category" {"code":2,"message":"Undefined array key \"category\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":191} - {"file":"Create.php","line":191,"function":"pickCategory","request-id":"64ad97c31543f","uid":"c0aa2c","process_id":2150242}
2023-07-11T17:58:59Z index [ERROR]: DB Error {"code":1054,"error":"Unknown column 'line-text' in 'field list'","callstack":"Database::selectToArray (112), Report::_selectOne (56), Report::selectOneById (98), Report::save (115), Create::post (228), BaseModule::run (707), App::runFrontend (52)","params":"SELECT `line-id`, `line-text` FROM `report-rule` WHERE (`rid` = 1)"} - {"file":"Database.php","line":706,"function":"p","request-id":"64ad986372b9b","uid":"cc069d","process_id":2166394}
2023-07-11T17:58:59Z index [WARNING]: E_WARNING: Undefined array key "uri-ids" {"code":2,"message":"Undefined array key \"uri-ids\"","file":"/var/www/html/mordor/src/Module/Moderation/Report/Create.php","line":341} - {"file":"Create.php","line":341,"function":"getAside","request-id":"64ad986392a61","uid":"a804d6","process_id":2168413}
2023-07-11T17:58:59Z index [ERROR]: Uncaught Exception TypeError: "count(): Argument #1 ($value) must be of type Countable|array, null given" at /var/www/html/mordor/src/Module/Moderation/Report/Create.php line 341 {"exception":"TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/mordor/src/Module/Moderation/Report/Create.php:341\nStack trace:\n#0 /var/www/html/mordor/src/Module/Moderation/Report/Create.php(341): count()\n#1 /var/www/html/mordor/src/Module/Moderation/Report/Create.php(139): Friendica\\Module\\Moderation\\Report\\Create->getAside()\n#2 /var/www/html/mordor/src/BaseModule.php(244): Friendica\\Module\\Moderation\\Report\\Create->content()\n#3 /var/www/html/mordor/src/App.php(707): Friendica\\BaseModule->run()\n#4 /var/www/html/mordor/index.php(52): Friendica\\App->runFrontend()\n#5 {main}"} - {"file":null,"line":null,"function":null,"request-id":"64ad986392a61","uid":"a804d6","process_id":2168413}
```

### Steps to Reproduce

1.  create a new report - I created on on a comment
2.  select 'unwanted behavior'
3.  leave an additional information
4. remove the 'forward to remote' option
5. hit the submit button

Actual Result:
WSOD

Expected Result:
Some success message and being redirected to the page where I came from.

### Screenshots
No. It's all white. :-)

### Platform Info

Friendica Version:
2023.09-dev
database version is 1522/1522, the post update version is 1507/1507

Friendica Source: 
git

PHP version:
8.1

SQL version:
10.6 MariaDB

**Comment by AlfredSK** [comment]
Ähm. There is no column `line-text`:

```
MariaDB [mordor]> desc `report-rule`;
+---------+------------------+------+-----+---------+-------+
| Field   | Type             | Null | Key | Default | Extra |
+---------+------------------+------+-----+---------+-------+
| rid     | int(10) unsigned | NO   | PRI | NULL    |       |
| line-id | int(10) unsigned | NO   | PRI | NULL    |       |
| text    | text             | NO   |     | NULL    |       |
+---------+------------------+------+-----+---------+-------+
```

**Comment by MrPetovan** [comment]
This is a PHP 8 error, see https://friendica.mrpetovan.com/display/735a2029-2264-ac23-1a2a-74a556437152

> [...] please replace line 341 of the src/Module/Moderation/Report/Create.php file by the following:
> ```
> 			'$posts'    => count($request['uri-ids'] ?? []),
> ```

**Comment by AlfredSK** [comment]
Ok. Thank you.

But what about the DB error for the missing column in table report-rule?

**Comment by MrPetovan** [comment]
That would be a bug in `src/Moderation/Repository/Report.php:112`, it should read
```php
		$reportRules = new Collection\Report\Rules(array_map([$this->ruleFactory, 'createFromTableRow'], $this->db->selectToArray('report-rule', ['line-id', 'text'], ['rid' => $condition['id'] ?? 0])));
```

**Comment by AlfredSK** [comment]
> This is a PHP 8 error, see https://friendica.mrpetovan.com/display/735a2029-2264-ac23-1a2a-74a556437152
> 
> > [...] please replace line 341 of the src/Module/Moderation/Report/Create.php file by the following:
> > ```
> > 			'$posts'    => count($request['uri-ids'] ?? []),
> > ```

@MrPetovan Can this change be made in develop? I mean is the proposed change backwards compatible with PHP 7.4? If so I would like to create a PR.

**Comment by MrPetovan** [comment]
Yes it is, please go ahead!

**Comment by AlfredSK** [comment]
Fixed by #13283
